src/coreclr/vm/jitinterface.cpp

```diff
@@ -4547,6 +4547,30 @@ bool CEEInfo::isExactType(CORINFO_CLASS_HANDLE cls)
     return result;
 }
 
+// Returns whether a class handle represents a generic type, if that can be statically determined.
+TypeCompareState CEEInfo::isGenericType(CORINFO_CLASS_HANDLE cls)
+{
+    CONTRACTL {
+        THROWS;
+        GC_TRIGGERS;
+        MODE_PREEMPTIVE;
+    } CONTRACTL_END;
+
+    TypeHandle typeHandle(cls);
+
+    TypeCompareState result = TypeCompareState::May;
+
+    JIT_TO_EE_TRANSITION();
+
+    if (typeHandle != TypeHandle(g_pCanonMethodTableClass))
+    {
+        result = typeHandle.HasInstantiation() ? TypeCompareState::Must : TypeCompareState::MustNot;
+    }
+
+    EE_TO_JIT_TRANSITION();
```

Comment:
```suggestion
    CONTRACTL {
        NOTHROW;
        GC_NOTRIGGER;
        MODE_PREEMPTIVE;
    } CONTRACTL_END;

    TypeHandle typeHandle(cls);

    TypeCompareState result = TypeCompareState::May;

    JIT_TO_EE_TRANSITION_LEAF();

    if (typeHandle != TypeHandle(g_pCanonMethodTableClass))
    {
        result = typeHandle.HasInstantiation() ? TypeCompareState::Must : TypeCompareState::MustNot;
    }

    EE_TO_JIT_TRANSITION_LEAF();
```

Reply:
You can update CEEInfo::isNullableType to same shape as well. ``_LEAF`` transition can be used for calls that do not trigger GC and throw exceptions. `_LEAF` makes the JIT/EE interface call cheaper and smaller.

Reply:
Also, `isNullableType` has the same `TypeHandle typeHandle = TypeHandle();` bug that was just fixed here. Could you please fix it there as well? The `if (typeHandle != TypeHandle(g_pCanonMethodTableClass))` check is actually unnecessary in `isNullable` - it would be fine to delete it.

Reply:
Applied all 3 suggested changes, thank you! 😄

Reply:
> The if (typeHandle != TypeHandle(g_pCanonMethodTableClass)) check is actually unnecessary in isNullable - it would be fine to delete it.

I presume it's indeed not needed since _Canon is never Nullable<> anyway, it explains why `TypeHandle();` didn't cause any issues